feat(color-picker): add multi-color selection & copy functionality

elibattistoni opened the pull request:
## Description

### Multi-Color Selection & Copy Functionality

Adds multi-color selection to the **"Organize Colors"** and **"Generate Colors"** commands with bulk copy capabilities.

### Features Added

#### Selection
- `Cmd + S` - Toggle individual color selection
- `Cmd + Shift + A` - Select all colors
- `Cmd + Shift + Z` - Clear selection
- Visual checkmarks (✓) for selected colors
- Works in both Organize Colors and Generate Colors commands

#### Copy Formats
- **JSON** - Structured data format
- **CSS Classes** - `.color-1 { color: #hex; }`  
- **CSS Variables** - `:root { --color-1: #hex; }`
- Access via `Cmd + Shift + Enter` or "Copy Selected Colors" submenu

## Screencast

<img width="2000" height="1250" alt="color-picker-4" src="https://github.com/user-attachments/assets/c2935db6-e850-4642-94df-ba538da59991" />
<img width="2000" height="1250" alt="color-picker-5" src="https://github.com/user-attachments/assets/4a21867f-c8e7-4ee7-a954-b9c5f47d8a8d" />
<img width="2000" height="1250" alt="color-picker-6" src="https://github.com/user-attachments/assets/c4930772-5c07-4a4b-b368-543626011155" />
<img width="2000" height="1250" alt="color-picker-7" src="https://github.com/user-attachments/assets/5b404571-2a3a-494c-92ef-a2d3b711a79d" />


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used in the `README` are located outside the metadata folder if they were not generated with our [metadata tool](https://developers.raycast.com/basics/prepare-an-extension-for-store#how-to-use-it)

LitoMore commented:
Hi @elibattistoni, sorry for the late response. I just tried your implementation, and I think we can have a better way to let users understand this functionality.

You can try using the [List](https://developers.raycast.com/api-reference/user-interface/list) to create a list view with checkbox icons.

Raycast also has [`List.Dropdown`](https://developers.raycast.com/api-reference/user-interface/list#list.dropdown) and [`Grid.Dropdown`](https://developers.raycast.com/api-reference/user-interface/grid#grid.dropdown) so that you can put the dropdown both on grid and list view and use the same state to manage.

Here is a pseudo-code:

```tsx
if (viewType === 'grid') {
	return (
		<List searchBarAccessory={
			<List.Dropdown tooltip="...">
				<List.Dropdown.Item title="Grid" value="grid" />
				<List.Dropdown.Item title="Multi-Select" value="multi-select" />
			</List.Dropdown>
		}>
			<List.Item title="Foo" value="foo" />
			<List.Item title="Bar" value="bar" />
		</List>
	);
}

return (
	<Grid searchBarAccessory={
		<Grid.Dropdown tooltip="...">
			<Grid.Dropdown.Item title="Grid" value="grid" />
			<Grid.Dropdown.Item title="Multi-Select" value="multi-select" />
		</Grid.Dropdown>
	}>
		<Grid.Item icon={Icon.Circle} title="Foo" value="foo" />
		<Grid.Item icon={Icon.CheckCircle} title="Foo" value="foo" />
	</Grid>
);
```

What do you think?

elibattistoni commented:
Hi @LitoMore thank you and I'm very sorry for the late reply, I caught a virus and I've been sick for two weeks.

I did not understand one thing: where would you suggest placing this List.Dropdown and Grid.Dropdown? In the "Organize Colors" and "Generate Colors" views, or in a brand new view that e.g. is opened from the Actions menu in the aforementioned views?

elibattistoni commented:
Hi @LitoMore — sorry for the long silence, life got in the way. I picked this back up: I refactored along the lines of your Oct 18 reference (Single/Multi mode toggle in `searchBarAccessory`, switching to a `<List>` view in multi mode) and reset the branch onto current `main` since the extension restructured significantly (Swift→Rust, `src/lib/`). Could you please re-open the PR? The branch has been force-pushed with the fresh implementation and updated screenshots. Thanks for the patient feedback!

elibattistoni commented:
Continued in #27420 (this thread is auto-closed and can't be re-opened from my side).
